csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -18,6 +18,10 @@ namespace ZeroC.Ice
     /// servants, identities, and proxies.</summary>
     public sealed class ObjectAdapter : IDisposable, IAsyncDisposable
     {
+        /// <summary>Returns the adapter ID of this object adapter, or the empty string if this object adapter does not
+        /// have an adapter ID.</summary>
+        public string AdapterId { get; }
```

Comment:
Wouldn't `null` make more sense for not set?

Reply:
I didn't change this logic, just the API. 

We could switch to nullable strings for adapter IDs and replica group IDs however this affects much more code and we'd have to decide how to handle empty adapter IDs / replica group IDs - either like null or as illegal values.